core/domain/rights_manager.py

```diff
@@ -215,6 +243,52 @@ def is_exploration_cloned(exploration_id):
     return bool(exploration_rights.cloned_from)
 
 
+def create_new_collection_rights(collection_id, committer_id):
+    collection_rights = ActivityRights(collection_id, [committer_id], [], [])
+    commit_cmds = [{'cmd': CMD_CREATE_NEW}]
+
+    collection_models.CollectionRightsModel(
+        id=collection_rights.id,
+        owner_ids=collection_rights.owner_ids,
+        editor_ids=collection_rights.editor_ids,
+        viewer_ids=collection_rights.viewer_ids,
+        community_owned=collection_rights.community_owned,
+        status=collection_rights.status,
+        viewable_if_private=collection_rights.viewable_if_private,
+    ).commit(committer_id, 'Created new collection', commit_cmds)
+
+    subscription_services.subscribe_to_collection(committer_id, collection_id)
+
+
+def get_collection_rights(collection_id):
+    """Retrieves the rights for this collection from the datastore."""
+    model = collection_models.CollectionRightsModel.get(collection_id)
+    if model is None:
+        raise Exception('This collection does not exist.')
+    return _get_activity_rights_from_model(model, ACTIVITY_TYPE_COLLECTION)
+
+
+def is_collection_private(collection_id):
+    collection_rights = get_collection_rights(collection_id)
+    return collection_rights.status == ACTIVITY_STATUS_PRIVATE
+
+
+def is_collection_public(collection_id):
+    collection_rights = get_collection_rights(collection_id)
+    return collection_rights.status == ACTIVITY_STATUS_PUBLIC
+
+
+def _get_activity_rights(activity_type, activity_id):
+    try:
+        if activity_type == ACTIVITY_TYPE_EXPLORATION:
+            return get_exploration_rights(activity_id)
+        elif activity_type == ACTIVITY_TYPE_COLLECTION:
+            return get_collection_rights(activity_id)
+    except Exception:
+        pass
+    return None
```

Comment:
> @seanlip: This can go in the previous line?
>> @BenHenning: Except in the invalid case when activity_type is neither ACTIVITY_TYPE_EXPLORATION nor ACTIVITY_TYPE_COLLECTION. It would technically return None anyway, but that seems a bit dirty to me. What do you think?

Oh, I see. OK, probably fine to leave as-is. Though, why don't we just throw an error if it's not either of the given types?